spring-cloud-aws-sqs/src/main/java/io/awspring/cloud/sqs/annotation/SqsListenerAnnotationBeanPostProcessor.java

```diff
@@ -44,12 +47,23 @@ public class SqsListenerAnnotationBeanPostProcessor extends AbstractListenerAnno
 
 	private static final String GENERATED_ID_PREFIX = "io.awspring.cloud.sqs.sqsListenerEndpointContainer#";
 
+	private final List<SqsListenerFilter> filters;
+
+	public SqsListenerAnnotationBeanPostProcessor(Optional<List<SqsListenerFilter>> filters) {
+		this.filters = filters.orElseGet(Collections::emptyList);
+	}
+
 	@Override
 	protected Class<SqsListener> getAnnotationClass() {
 		return SqsListener.class;
 	}
 
+	@Override
 	protected Endpoint createEndpoint(SqsListener sqsListenerAnnotation) {
+		if (filters.stream().anyMatch(f -> !f.createEndpoint(sqsListenerAnnotation))) {
```

Comment:
I believe this logic would be better placed in the `AbstractListenerAnnotationBeanPostProcessor`, since it's not an SQS-specific logic.